### ISSUE TITLE
GUI crashes after setting layouts and closing a 3D image

### ISSUE THREAD

**Issue by durga2112** [Astroua/CARTAvis#84]
Thank randomized py.test runs for finding the particular sequence of actions that lead to this bug. Here's what to do:

- after launching the GUI, close all open images
- load a 3D image/data cube
- using the animator, set the channel to something other than 0
- set the layout to Image
- set the layout back to Analysis
- close the image

At this point, the GUI crashes for me every time. I put some debug statements in the code and I think the problem is happening when `_getImagePt()` is being called from `_getCursorText()`, but I don't know enough about what's going on there to fix it.

**Comment by slovelan** [comment]
I have been unable to reproduce this bug in my local build.  What code
base are you running against?  What image are you opening?  Are you
doing some of this through the script client rather than the GUI? If so,
could you please submit the script you are running.

**Comment by durga2112** [comment]
I'm running against what's in the jeff branch right now - so basically whatever is in master plus some of my own recent changes. It is entirely possible that you have already fixed this issue, so once your pull request has been accepted I can come back to it and if it's still a problem, I'll let you know.

While I initially discovered this bug through the use of scripted commands, I have been able to reproduce it entirely within the GUI, so no scripted commands are necessary. I don't think the specific data cube matters, because I have been able to reproduce it with different cubes. I do notice that the crash does not occur sometimes, but this is very infrequently - I'd say it crashes at least 90% of the time. Also, another thing I have noticed is that after switching from Image to Analysis layout, the animator shows the channel as 0, even though what's shown in the image viewer is not channel 0.
